src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -1488,10 +1488,42 @@ private fun processLexicalDeclarations(
         }
     }
 
-    fun processCondition(condition: RsCondition?, processor: RsResolveProcessor): Boolean {
-        if (condition == null || condition == cameFrom) return false
-        val pat = condition.pat ?: return false
-        return processPattern(pat, processor)
+    fun processLetExprs(
+        expr: RsExpr?,
+        processor: RsResolveProcessor,
+        prevScope: MutableMap<String, Set<Namespace>> = hashMapOf()
+    ): Boolean {
+        if (expr == null || expr == cameFrom) return false
+
+        val shadowingProcessor = createProcessor(processor.names) { e ->
+            (e.name !in prevScope) && processor(e).also {
+                if (processor.acceptsName(e.name)) {
+                    prevScope[e.name] = VALUES
+                }
+            }
+        }
+
+        fun process(expr: RsExpr): Boolean {
+            when (expr) {
+                cameFrom -> return false
+
+                is RsLetExpr -> {
+                    if (expr.expr == cameFrom) return false
```

Comment:
Hmm, is `expr == cameFrom` or `expr.expr == cameFrom` ever possible? is seems like `cameFrom` is filtered out before calling `processLetExprs`